clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -467,6 +468,7 @@ boolean joinGroupIfNeeded(final Timer timer) {
                 final RuntimeException exception = future.exception();
 
                 resetJoinGroupFuture();
+                rejoinReason = "rebalance failed due to " + exception.getClass() + " error: " + exception.getMessage();
```

Comment:
Example on the broker side:
```
[2021-12-17 11:29:16,061] INFO [GroupCoordinator 0]: Preparing to rebalance group test in state PreparingRebalance with old generation 1 (__consumer_offsets-48) (reason: Adding new member console-consumer-1d5a9905-c271-4700-a817-62fc9b9f28fc with group instance id None. Member joined due to rebalance failed due to class org.apache.kafka.common.errors.MemberIdRequiredException error: The group member needs to have a valid member id before actually entering a consumer group.) (kafka.coordinator.group.GroupCoordinator)
```

* Should we only get the `getSimpleName` of the class?
* There are many `:` in the log. I wonder if we could remove the one we've put here. Perhaps, we could use the following pattern: `rebalance failed due to '$message' ($class)`. What do you think?

Reply:
+1 for `getSimpleName` for the class. In addition to the David's suggestion, I think we should also remove the `due to`, because there is already 1 `due to` in the sentence. ex:  `rebalance failed: '$message' ($class)`

Reply:
thanks for the suggestions. i'll go ahead with david's suggestion as we're updating `Member joined due to` to `client reason:`